.github/workflows/bcny-cmake.yml

```diff
@@ -6,44 +6,95 @@ on:
 jobs:
   build:
     runs-on: swift-build-windows-latest-32-cores
+    outputs:
+      artifact_amd64: ${{ steps.vars.outputs.artifact_amd64 }}
+      artifact_arm64: ${{ steps.vars.outputs.artifact_arm64 }}
+      release: ${{ steps.vars.outputs.release }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - arch: 'amd64'
+            cmake_arch: 'AMD64'
+            cpack_arch: 'x64'
+          - arch: 'arm64'
+            cmake_arch: 'ARM64'
+            cpack_arch: 'arm64'
 
     steps:
     - uses: compnerd/gha-setup-vsdevenv@main
+      with:
+        host_arch: amd64
+        components: 'Microsoft.VisualStudio.Component.VC.Tools.x86.x64;Microsoft.VisualStudio.Component.VC.Tools.ARM64'
+        arch: ${{ matrix.arch }}
     - uses: actions/checkout@v4
       with:
         path: ${{ github.workspace }}/SourceCache/cmake
-    - run: cmake -B ${{ github.workspace }}/BinaryCache/cmake -G Ninja -S ${{ github.workspace }}/SourceCache/cmake -D CMAKE_BUILD_TYPE=Release
+    - run: >
+        cmake -B ${{ github.workspace }}/BinaryCache/cmake `
+              -G Ninja `
+              -S ${{ github.workspace }}/SourceCache/cmake `
+              -D CMAKE_BUILD_TYPE=Release `
+              -D CMAKE_SYSTEM_NAME=Windows `
+              -D CMAKE_SYSTEM_PROCESSOR=${{ matrix.cmake_arch }}
     - run: cmake --build ${{ github.workspace }}/BinaryCache/cmake
     - run: |
         cd ${{ github.workspace }}/BinaryCache/cmake
-        cpack -G WIX
-    - uses: actions/upload-artifact@v3
+        cpack -G WIX -D CPACK_SYSTEM_NAME=win64-${{ matrix.cpack_arch }} -D CPACK_TOPLEVEL_TAG=win64-${{ matrix.cpack_arch }}
+    - uses: actions/upload-artifact@v4
       with:
-        name: cmake-windows-amd64
+        name: cmake-windows-${{ matrix.arch }}
         path: ${{ github.workspace }}/BinaryCache/cmake/*.msi
     - id: vars
       run: |
         $Msi = Get-ChildItem ${{ github.workspace }}/BinaryCache/cmake/*.msi | Select-Object -First 1
         $Artifact = $Msi.Name
-        $Location = $Msi.FullName
-        $Release = [System.IO.Path]::GetFileNameWithoutExtension($Artifact)
-        Write-Output "artifact=$($Artifact)" >> $Env:GITHUB_OUTPUT
-        Write-Output "location=$($Location)" >> $Env:GITHUB_OUTPUT
+        $Filename = [System.IO.Path]::GetFileNameWithoutExtension($Artifact)
+        # Drop the arch string
+        $Release = $Filename.Remove($Filename.LastIndexOf('-'))
```

Comment:
Given this job does not use `always()` if one of the build matrix jobs fail then this will not run. So even more fail fast being false makes little sense on the surface.

Reply:
@compnerd